types/cordova-plugin-safariviewcontroller/index.d.ts

```diff
@@ -0,0 +1,214 @@
+     * @param onSuccess The callback function to call after successfully setting the custom tab implementation.
+     * @param onError The callback function to call in case of an error or if method is unsupported on current platform.
+     */
+    useCustomTabsImplementation(
+        packageName: string,
+        onSuccess?: (result: true) => void,
+        onError?: (error: unknown) => void,
+    ): void;
+
+    /**
+     * Try to connect to the Chrome's custom tabs service. You must call this method before calling
+     * `warmUp()` or `mayLaunchUrl()` methods.
+     *
+     * **This method Android-specific. It will always error out on other platforms.**
+     *
+     * @param onSuccess The callback function to call after successfully connecting to the custom tabs service.
+     * @param onError The callback function to call in case of an error or if method is unsupported on current platform.
+     */
+    connectToService(onSuccess?: (result: true) => void, onError?: (error: unknown) => void): void;
+
+    /**
+     * Warm up the browser process. Call this method whenever there's a chance the user will open an external url. See
+     * [Custom Tabs implementation guide](https://developers.google.com/web/android/custom-tabs/implementation-guide#warm_up_the_browser_process)
+     * for more details.
+     *
+     * **This method Android-specific. It will always error out on other platforms.**
+     *
+     * @param onSuccess The callback function to call after successfully warming up the browser process.
+     * @param onError The callback function to call in case of an error or if method is unsupported on current platform.
+     */
+    warmUp(onSuccess?: (result: true) => void, onError?: (error: unknown) => void): void;
+
+    /**
+     * Prepare the selected custom tab implementation to navigate to passed URL. For even better performance optimization,
+     * call this methods if there's more than a 50% chance the user will open a certain URL. See
+     * [CustomTabsSession docs](https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,%20android.os.Bundle,%20java.util.List%3Candroid.os.Bundle%3E))
+     * for more details.
+     *
+     * **This method Android-specific. It will always error out on other platforms.**
+     *
+     * @param url The URL that should be loaded next by the custom tab.
+     * @param onSuccess The callback function to call after custom tab has accepted the URL to launch.
+     * @param onError The callback function to call in case custom tab hasn't accepted the URL to launch,
+     *                if an error has occurred, or if method is unsupported on current platform.
+     */
+    mayLaunchUrl(url: string, onSuccess?: (result: true) => void, onError?: (error: unknown) => void): void;
+}
+
+declare var SafariViewController: SafariViewController;
```

Comment:
looking at the source code (https://unpkg.com/browse/cordova-plugin-safariviewcontroller@2.0.0/www/SafariViewController.js), it exports things as `module.exports = { isAvailable: .... }`.
is there a reason why we shouldn't be matching this in the d.ts file and doing something like `export = ...` here instead?

Reply:
It's because of how cordova plugins work. I you look into the plugin's [`plugin.xml`](https://github.com/EddyVerbruggen/cordova-plugin-safariviewcontroller/blob/master/plugin.xml), you'll see the following definition:
```xml
  <js-module name="SafariViewController" src="www/SafariViewController.js">
    <clobbers target="SafariViewController"/>
  </js-module>
```

This means that the module will be exposed in the global scope as `SafariViewController`.